watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/dye_desalination/dye_desalination.py

```diff
@@ -148,71 +215,218 @@ def eq_flow_mass_comp(blk, j):
                     == blk.properties_out[0].flow_mass_phase_comp["Liq", "H2O"]
                 )
 
+    elif include_dewatering and include_gac:
+        raise TypeError(
+            "This system is not designed to have both dewatering and GAC units."
+        )
+    else:
+        m.fs.concentrated_dye = Product(property_package=m.fs.prop_nf)
+
+    # pretreatment
+    if hasattr(m.fs, "pretreatment"):
+        prtrt.wwtp = SecondaryTreatmentWWTPZO(
+            property_package=m.fs.prop_nf, database=m.db, process_subtype="default"
+        )
+    else:
+        pass
+
+    # nanofiltration components
+    dye_sep.P1 = PumpElectricityZO(
+        property_package=m.fs.prop_nf, database=m.db, process_subtype="default"
+    )
+
+    dye_sep.nanofiltration = NanofiltrationZO(
+        property_package=m.fs.prop_nf,
+        database=m.db,
+        process_subtype="rHGO_dye_rejection",
+    )
+
+    # reverse osmosis components
+    if include_RO:
+        desal = m.fs.desalination = Block()
+        desal.P2 = Pump(property_package=m.fs.prop_ro)
+        if RO_1D:
+            desal.RO = ReverseOsmosis1D(
+                property_package=m.fs.prop_ro,
+                has_pressure_change=True,
+                pressure_change_type=PressureChangeType.calculated,
+                mass_transfer_coefficient=MassTransferCoefficient.calculated,
+                concentration_polarization_type=ConcentrationPolarizationType.calculated,
+            )
+        else:
+            desal.RO = ReverseOsmosis0D(
+                property_package=m.fs.prop_ro,
+                has_pressure_change=True,
+                pressure_change_type=PressureChangeType.calculated,
+                mass_transfer_coefficient=MassTransferCoefficient.calculated,
+                concentration_polarization_type=ConcentrationPolarizationType.calculated,
+            )
+
+        desal.RO.width.setub(2000)
+        desal.RO.area.setub(20000)
```

Comment:
Do we need this?

Reply:
This appears to be necessary (i.e. deleting it results in the solver failing to return an optimal solution)